iroh-sync/src/store.rs

```diff
@@ -132,6 +140,24 @@ pub trait Store: std::fmt::Debug + Clone + Send + Sync + 'static {
     /// Get all content hashes of all replicas in the store.
     fn content_hashes(&self) -> Result<Self::ContentHashesIter<'_>>;
 
+    /// Get the latest timestamp and key for each author in a namespace
+    fn get_latest(&self, namespace: NamespaceId) -> Result<Self::LatestIter<'_>>;
+
+    /// Check if a state vector contains pointers that we do not have locally.
+    fn has_news_for_us(&self, namespace: NamespaceId, heads: &AuthorHeads) -> Result<bool> {
+        let our_heads = {
+            let latest = self.get_latest(namespace)?;
+            let mut heads = AuthorHeads::default();
+            for e in latest {
+                let (author, timestamp, _key) = e?;
+                heads.insert(author, timestamp);
+            }
+            heads
+        };
+        let has_news_for_us = heads.has_news_for(&our_heads);
+        Ok(has_news_for_us)
```

Comment:
it might be interesting (at least for tracking purposes) to not just return `true` or `false`, but rather the number of keys that we have news for

Reply:
Good idea. Changed it to return `Option<NonZeroU64>`. However it returns the number of *authors*, not keys, that there's updates for.